web/src/app/modules/shared/components/presentation/json-editor/json-editor.component.ts

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright (c) 2021 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+import {
+  AfterViewChecked,
+  ChangeDetectionStrategy,
+  Component,
+} from '@angular/core';
+import { AbstractViewComponent } from '../../abstract-view/abstract-view.component';
+import JSONEditor from 'jsoneditor';
+import { JSONEditorView } from '../../../models/content';
+
+@Component({
+  selector: 'app-view-json',
+  templateUrl: 'json-editor.component.html',
+  styleUrls: ['./json-editor.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class JSONEditorComponent
+  extends AbstractViewComponent<JSONEditorView>
+  implements AfterViewChecked {
+  container: HTMLElement;
+  id: string;
+  options = {
+    mode: '',
+    mainMenuBar: false,
+    navigationBar: false,
+  };
+  content: any;
+
+  constructor() {
+    super();
+    this.id = Math.random().toString(36).substring(2, 15);
+  }
+
+  ngAfterViewChecked() {
+    this.container = document.getElementById(this.id);
+    if (this.container) {
+      const editor = new JSONEditor(this.container, this.options);
+      editor.set(this.content);
+    }
+  }
+
+  update() {
+    const view = this.v;
+    this.options.mode = view.config.mode;
+
+    try {
+      this.content = JSON.parse(view.config.content);
+    } catch (e) {
+      this.options.mode = 'preview';
+      this.content = {
+        error: 'cannot parse json',
```

Comment:
Not a huge issue, just wondering if this could be tested.

Reply:
It is now tested